### PR TITLE
chore(exchange-io-erc1888): Move supply watching functionality from e…

### PR DESCRIPTION
**Pull request by arvindchez**
…rc1888 watcher exchange using eventbus

### CODE REVIEW

**Comment by kosecki123** [comment]
Based on the changes this PR does 2 things:
- declares events in the `exchange` package
- moved code to handlers in the `exchange-io-erc1888`
but this does not achieve the goal of the PR which is to move the "logic" of post for sale from `exchange-io-erc1888` to the `exchange` package. Note that post for sale is not dependent on the method of "discovering" the deposit thus can be reused in case of other `exchange-io-*` packages.

**Review by kosecki123 — CHANGES_REQUESTED** [review]
As commented in the PR.

**Review by josipbagaric — COMMENTED** [review]
So in order for this to work we need to:
1.  Move the event handlers to the `@energyweb/exchange` packages
2. Move `this.tryPostForSale()` functionality from `@energyweb/exchange-io-erc1888` to `@energyweb/exchange` 

The deposit watcher should only trigger the `DepositDiscoveredEvent` event and the rest should be handled by the exchange.

Did I get this correctly @kosecki123?

File: packages/exchange/src/pods/post-for-sale/handlers/deposit-discovered-event.handler.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    public async handle(event: DepositDiscoveredEvent): Promise<void> {
+        const { address, amount, asset } = event.deposit;
+        this.logger.debug(
+            `Deposit discovered event raised for deposit with tokenId=${asset.tokenId} from=${address} with value=${amount}`
+        );
+
+        this.transferService.storeDeposit(event.deposit);
+    }
+}
```
Comment (kosecki123) [inline packages/exchange/src/pods/post-for-sale/handlers/deposit-discovered-event.handler.ts]: This file should be in `transfer` pod. Storing deposit has nothing to do with posting for sale.

File: packages/exchange-io-erc1888/src/deposit-watcher/deposit-watcher.service.ts [hunk 2]
```diff
@@ -2,22 +2,20 @@ import { Contracts } from '@energyweb/issuer';
...
 
 import {
     IExchangeConfigurationService,
-    IExternalDeviceService,
-    AccountService,
-    CreateAskDTO,
-    OrderService,
-    TransferStatus,
-    TransferService
+    TransferService,
```
Comment (kosecki123) [inline packages/exchange-io-erc1888/src/deposit-watcher/deposit-watcher.service.ts:12]: This can be removed I guess.

**Review by kosecki123 — CHANGES_REQUESTED** [review]
Looks good 👍 2 changes request in the comments.

File: packages/exchange-io-erc1888/src/deposit-watcher/deposit-watcher.service.ts (hunk 2 above)
Comment (arvindchez) [reply packages/exchange-io-erc1888/src/deposit-watcher/deposit-watcher.service.ts:12]: The transfer service is needed for below code block. i.e to get the confirmation block. Please confirm.    

 const blockNumber = await this.transferService.getLastConfirmationBlock();

        this.logger.debug(`Starting from block ${blockNumber}`);

        this.provider.resetEventsBlock(blockNumber);
        this.provider.on(
            {
                address: this.registryAddress,
                topics
            },
            (event: providers.Log) => this.processEvent(event)
        );

**Review by josipbagaric — CHANGES_REQUESTED** [review]
The PR doesn't follow our commit scheme. Please rebase the PR and make sure that all commit descriptions match our commit scheme.

**Comment by arvindchez** [comment]
Send out a new clean PR (https://github.com/energywebfoundation/origin/pull/1932) with all the changes as per comments here.

**Pull request closed without merging**